### PR TITLE
[feat] view plants: created PlantCard component

### PR DESCRIPTION
**Pull request by SashankBalusu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
The main changes were adding the plant card component and supabase function call (the function that is called in this file is also viewable on supabase)
- the styling for the plant card components is still rough as the design is changing + moving to styled components will be a later sprint task, so the important stuff is mainly the html 

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

Screenshot 2024-10-08 at 5 44 23 PM

Rough styling --still need to add icons, etc

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- for supabase functions, the order is probably to look at the supabase function in the actually database and then the syntax here. 
- for plant card component again the styling will be changed, but would appreciate feedback on the html structure

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- styling as mentioned a couple times before

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
https://stackoverflow.com/questions/55688418/what-is-a-non-setof-function-in-postgresql

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: components/PlantCard/PlantCard.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import styles from './PlantCardStyles.module.css';
+
+function format(start: String, end: String) {
```
Comment (ccatherinetan) [inline components/PlantCard/PlantCard.tsx]: are we using this function anywhere? if not, we can remove it?

File: components/PlantCard/PlantCard.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December',
+  ];
+}
+export default function PlantCard(plantObj: Plant) {
```
Comment (ccatherinetan) [inline components/PlantCard/PlantCard.tsx]: make sure you include the definition of the `Plant` type in your code; otherwise this will error! 

try running `git rebase main` → the Plant type should already be defined on main!

**Review by ccatherinetan — COMMENTED** [review]
Looking great so far! Would it be possible to create an instance of this component in the `app/view-plants` directory to test it out? 
Also make sure to run `npm run lint` and `npm run prettier:fix` before you push to pass the styling checks!

File: components/PlantCard/PlantCardStyles.module.css [hunk 3]
```diff
@@ -0,0 +1,52 @@
+.Card {
```
Comment (ccatherinetan) [inline components/PlantCard/PlantCardStyles.module.css:1]: thanks so much for starting the initial styling! can we migrate these styles to Styled Components? 
see [kevin's](https://github.com/calblueprint/trap-garden/pull/6) pr for reference on using Styled Components! 
Styled components doc: https://styled-components.com/
Here's a `styles.tsx` file reference from IJP: https://github.com/calblueprint/immigration-justice-project/blob/main/src/components/ListingDetails/styles.ts

File: components/PlantCard/PlantCard.tsx (hunk 1 above)
Comment (SashankBalusu) [reply components/PlantCard/PlantCard.tsx]: oh shoot that was a remnant of an older time will remove now

File: components/PlantCard/PlantCardStyles.module.css [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: start;
+  border-radius: 12px;
+  background-color: white;
+  box-shadow:
+    0 24px 38px 3px rgba(0, 0, 0, 0.14),
+    0 9px 46px 8px rgba(0, 0, 0, 0.12),
+    0 11px 15px -7px rgba(0, 0, 0, 0.2);
+}
```
Comment (kylezryr) [inline components/PlantCard/PlantCardStyles.module.css:13]: image
The text goes outside of the div depending on the browser zoom, maybe consider setting an 'overflow' attribute? or setting a fixed font size for the text since it currently changes depending on the zoom

File: components/PlantCard/PlantCardStyles.module.css [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  height: 60%;
+  width: 100%;
+  background-color: #f5f6f6;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  border-top-left-radius: 12px;
+  border-top-right-radius: 12px;
+}
+.cardContent {
```
Comment (kylezryr) [inline components/PlantCard/PlantCardStyles.module.css]: This would probably change anyways when you switch to styled components but style class names should use Pascal Case (e.g. CardContent) instead of Camel Case (but this is a small thing lol)

File: src/api/supabase/createClient.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
+import { createClient } from '@supabase/supabase-js';
```
Comment (ccatherinetan) [inline src/api/supabase/createClient.ts]: make sure to delete this file as the createClient file already exists in the `api/supabase` route

File: src/api/supabase/queries/plant_by_id.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import { UUID } from 'crypto';
```
Comment (ccatherinetan) [inline src/api/supabase/queries/plant_by_id.ts]: this file is also duplicated in the `api/supabase/queries` route. let's delete the entire `src/` directory!

File: app/view-plants/page.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
+import PlantCard from '@/components/PlantCard/PlantCard';
+import { Plant } from '@/types/schema';
+
+export default function Home() {
+  const plant: Plant = {
```
Comment (ccatherinetan) [inline app/view-plants/page.tsx]: can we also try using the `get_plant_by_id` query you wrote here to display the PlantCard!

File: components/PlantCard/PlantCard.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import { Plant } from '@/types/schema';
+import styles from './PlantCardStyles.module.css';
+
+export default function PlantCard(plantObj: Plant) {
```
Comment (ccatherinetan) [inline components/PlantCard/PlantCard.tsx]: I think this is what's causing the syntax error. The way you're passing in the plantObj now makes it seem like you're passing the individual fields in plant as props, which is why you have to use the spread operator in `view-plants` (`<PlantCard {...plant}><PlantCard/>`). 

However, if we want to pass the entire Plant object, we should define PlantCard like this: 
`PlantCard({ plant }: { plant: Plant })`
Then, in `view-plants`, you can use the component like this: `<PlantCard plantObj={plant}/>`

Comment (ccatherinetan) [reply components/PlantCard/PlantCard.tsx]: also, nit, but i think we can rename the `plantObj` prop to just `plant`

**Comment by ccatherinetan** [comment]
great work! i think the final changes are 
1. Removing the `src/`directory. 
2. Fixing the way we're passing props for `PlantCard` (see my comment; we prob don't want to use the spread operator) 
3. Try to actually test your `get_plant_by_id` query to display a PlantCard in `view-plants`
Kyle left some great comments about the styling; if you have time that would be great to look at, but we're prioritizing backend functionality first. So if you could address these 3 things, we can merge it in!

File: api/supabase/queries/plant_by_id.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import { UUID } from 'crypto';
+import { Plant } from '@/types/schema';
+import supabase from '../createClient';
+
+export async function get_plant_by_id(
```
Comment (ccatherinetan) [inline api/supabase/queries/plant_by_id.ts]: nit: let's also rename this to use camelCase `getPlantById`
